### ISSUE TITLE
Learning rate isn't updated with 2 schedulers for 1 optmizer

### ISSUE THREAD

**Issue by davidnvq** [PyTorchLightning/pytorch-lightning#8884]
## 🐛 Bug
When I specify 2 learning rate schedulers for 1 optimizer, the learning rate is only controlled by the first scheduler.

### To Reproduce

https://colab.research.google.com/drive/1K31mLwat1sGsd1HUcayiir9c0YQDUP0T?usp=sharing

```py
def warmup_lr_scheduler(optimizer, warmup_iters, warmup_factor):
    def f(x):
        if x >= warmup_iters:
            return 1
        alpha = float(x) / warmup_iters
        return warmup_factor * (1 - alpha) + alpha
    return torch.optim.lr_scheduler.LambdaLR(optimizer, f)

# inside the `pl.LightningModule`:
    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.layer.parameters(), lr=0.1)
        # The lr will increase from lr/100 -> lr in the first 100 iterations
        warmup_scheduler = {
            'scheduler': warmup_lr_scheduler(optimizer, 100, 1. / 100),
            'name': 'learning_rate',
            'interval':'step',
            'frequency': 1
        }
        lr_scheduler = {
            'scheduler': torch.optim.lr_scheduler.MultiStepLR(optimizer, milestones=[2, 4, 8], gamma=0.1),
            'name': 'learning_rate',
            'interval':'epoch',
            'frequency': 1
        }

        return [optimizer], [warmup_scheduler, lr_scheduler]
```

### Expected behavior
* Learning rate will increase from `lr/100` to `lr` in the first 100 steps. **(it is expected)**
* After that, learning rate remains the same as `lr`, although it is supposed to decrease after epoch 2, 4, 8. **(seems to have bugs here)**

### Environment

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.18.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.6.0+cu101
	- pytorch-lightning: 0.10.0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020

**Comment by davidnvq** [comment]
It was a bug from my side that I should run `warmup_scheduler` only in the first epoch. However, it is not trivial do to so as I haven't figured it out.
A tentative solution is done by following [the example](https://pytorch-lightning.readthedocs.io/en/stable/common/optimizers.html?highlight=learning%20rate%20scheduler#learning-rate-scheduling) for warming up lr: 
```python

    # learning rate warm-up
    def optimizer_step(
        self,
        epoch,
        batch_idx,
        optimizer,
        optimizer_idx,
        optimizer_closure,
        on_tpu=False,
        using_native_amp=False,
        using_lbfgs=False,
    ):
        # skip the first 100 steps
        if self.trainer.global_step <= 100:
            warmup_iters, warmup_factor = 100, 1/100.
            alpha = float(self.trainer.global_step) / warmup_iters
            lr_scale = warmup_factor * (1 - alpha) + alpha
            for pg in optimizer.param_groups:
                pg["lr"] = lr_scale * 0.1

        # update params
        optimizer.step(closure=optimizer_closure)
     def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.layer.parameters(), lr=0.1)
        
        lr_scheduler = {
            'scheduler': torch.optim.lr_scheduler.MultiStepLR(optimizer, milestones=[2, 4, 6, 8], gamma=0.1),
            'name': 'learning_rate',
            'interval':'epoch',
            'frequency': 1
        }
        return [optimizer], [lr_scheduler]
```

Still, is there any better way to handle this?

**Comment by carmocca** [comment]
> Still, is there any better way to handle this?

I think that's entirely fine. Closing the issue as you've found the cause 😃
